exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/DrillFilterJoinRules.java

```diff
@@ -59,20 +58,44 @@ public boolean apply(Join join, JoinRelType joinType, RexNode exp) {
         }
       };
 
+  /** Predicate that always returns true for any filter in OUTER join, and only true
+   * for strict EQUAL or IS_DISTINCT_FROM conditions (without functions) over RexInputRef in INNER join.
+   * With this predicate, the filter expression that return true will be kept in the JOIN OP.
+   * Example:  INNER JOIN,   L.C1 = R.C2 will be kepted in JOIN.
+   *                         L.C3 + 100 = R.C4 + 100 and L.C5 < R.C6 will be pulled up into Filter above JOIN.
+   *           OUTER JOIN,   Keep any filter in JOIN.
+  */
+  public static final FilterJoinRule.Predicate STRICT_EQUAL_IS_DISTINCT_FROM =
```

Comment:
It's not clear what the meaning of STRICT_EQUAL is here ?  Does the expression  a + 1 = b + 1 qualify ?

Reply:
Yeah, basically STRICT_EQUAL means only a = b types of predicates.